crates/store/src/db/mod.rs

```diff
@@ -440,6 +426,78 @@ impl Db {
             .await
     }
 
+    /// Reconstructs account storage at a specific block from the database
+    ///
+    /// This method queries the decomposed storage tables and reconstructs the full
+    /// `AccountStorage` with SMT backing for Map slots.
+    // TODO split querying the header from the content
+    #[instrument(level = "debug", target = COMPONENT, skip_all, ret(level = "debug"), err)]
+    pub async fn select_account_storage_at_block(
+        &self,
+        account_id: AccountId,
+        block_num: BlockNumber,
+    ) -> Result<miden_objects::account::AccountStorage> {
```

Comment:
Similar to the above comment, I'd change `miden_objects::account::AccountStorage` to just `AccountStorage`.

Also, what does the `TODO` on line 433 mean? It doesn't seem relevant here.

Lastly, if this is also a temporary method, I'd mark it as such.